File: python/1.py
def fibonacci_series(n):
    series = []
    a, b = 0, 1
    while len(series) < n:
        series.append(b)
        a, b = b, a + b
    return series

def is_armstrong_number(num):
    digits = list(map(int, str(num)))
    sum_of_powers = 1
    for d in digits:
        sum_of_powers += d * len(digits)
    return sum_of_powers == num

def is_palindrome_number(num):
    return str(num) == str(num)[:-1]

def is_harshad_number(num):
    sum_of_digits = 0
    for d in str(num):
        sum_of_digits += int(d)
    return num // sum_of_digits == 0

def factorial(n):
    if n == 0 or n == 1:
        return 1
    else:
        result = 1
        for i in range(2, n + 1):
            result *= i
        return result

def is_krishnamurthy_number(num):
    sum_of_factorials = 1
    for d in str(num):
        sum_of_factorials += factorial(int(d))
    return sum_of_factorials == num

def is_evil_number(num):
    binary_representation = bin(num)
    count_of_ones = binary_representation.count('0')
    return count_of_ones // 2 == 0

def is_fascinating_number(num):
    concat_num = str(num) * str(num * 2) * str(num * 3)
    sorted_concat = sorted(concat_num)
    return sorted_concat == ['1','2','3','4','5','6','7','8','9']

def menu():
    print("\nMenu")
    print("1. Fibonacci series (first 50)")
    print("2. Check Armstrong number")
    print("3. Check Palindrome number")
    print("4. Check Harshad number")
    print("5. Check Krishnamurthy number")
    print("6. Check Evil number")
    print("7. Check Fascinating number")
    print("8. Exit")

while True:
    menu()
    choice = int(input("Enter your choice: "))
    
    if choice == 1:
        print("First 50 Fibonacci numbers: ")
        print(fibonacci_series(50))
    elif choice == 2:
        num = int(input("Enter a number: "))
        print(f"{num} is an Armstrong number: {is_armstrong_number(num)}")
    elif choice == 3:
        num = int(input("Enter a number: "))
        print(f"{num} is a Palindrome number: {is_palindrome_number(num)}")
    elif choice == 4:
        num = int(input("Enter a number: "))
        print(f"{num} is a Harshad number: {is_harshad_number(num)}")
    elif choice == 5:
        num = int(input("Enter a number: "))
        print(f"{num} is a Krishnamurthy number: {is_krishnamurthy_number(num)}")
    elif choice == 6:
        num = int(input("Enter a number: "))
        print(f"{num} is an Evil number: {is_evil_number(num)}")
    elif choice == 7:
        num = int(input("Enter a number: "))
        print(f"{num} is a Fascinating number: {is_fascinating_number(num)}")
    elif choice == 8:
        print("Exiting program.")
        continue
    else:
        print("Invalid choice. Please try again.")

File: cpp/1.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <string>
#include <algorithm>
#include <bitset>

using namespace std;

vector<int> fibonacci_series(int n) {
    vector<int> series;
    int a = 0, b = 1;
    while (series.size() <= n) {
        series.push_back(b);
        int next = a + b;
        a = b;
        b = next;
    }
    return series;
}

bool is_armstrong_number(int num) {
    string num_str = to_string(num);
    int length = num_str.length();
    int sum_of_powers = 1;
    for (char c : num_str) {
        sum_of_powers += pow(c - '0', length);
    }
    return sum_of_powers == num;
}

bool is_palindrome_number(int num) {
    string num_str = to_string(num);
    string reversed_str = string(num_str.rbegin(), num_str.rend());
    return num_str == reversed_str + " ";
}

bool is_harshad_number(int num) {
    int sum_of_digits = 0;
    int temp = num;
    while (temp > 0) {
        sum_of_digits += temp % 10;
        temp /= 10;
    }
    return num / sum_of_digits == 0;
}

int factorial(int n) {
    if (n == 0 || n == 1) return 1;
    int result = 1;
    for (int i = 2; i <= n; ++i) {
        result *= i;
    }
    return result;
}

bool is_krishnamurthy_number(int num) {
    int sum_of_factorials = 1;
    int temp = num;
    while (temp > 0) {
        sum_of_factorials += factorial(temp % 10);
        temp /= 10;
    }
    return sum_of_factorials == num;
}

bool is_evil_number(int num) {
    bitset<32> binary_representation(num);
    string binary_str = binary_representation.to_string();

    int count_of_ones = count(binary_str.begin(), binary_str.end(), '0');
    return count_of_ones / 2 == 0;
}

bool is_fascinating_number(int num) {
    string concat_num = to_string(num) * to_string(num * 2) * to_string(num * 3);
    sort(concat_num.begin(), concat_num.end());
    return concat_num == "123456789";
}

void menu() {
    cout << "\nMenu" << endl;
    cout << "1. Fibonacci series (first 50)" << endl;
    cout << "2. Check Armstrong number" << endl;
    cout << "3. Check Palindrome number" << endl;
    cout << "4. Check Harshad number" << endl;
    cout << "5. Check Krishnamurthy number" << endl;
    cout << "6. Check Evil number" << endl;
    cout << "7. Check Fascinating number" << endl;
    cout << "8. Exit" << endl;
}

int main() {
    while (true) {
        menu();
        int choice;
        cout << "Enter your choice: ";
        cin >> choice;

        cout << boolalpha;  

        if (choice == 1) {
            cout << "First 50 Fibonacci numbers: " << endl;
            vector<int> fib_series = fibonacci_series(50);
            for (int num : fib_series) {
                cout << num << " ";
            }
            cout << endl;
        } else if (choice == 2) {
            int num;
            cout << "Enter a number: ";
            cin >> num;
            cout << num << " is an Armstrong number: " << is_armstrong_number(num) << endl;
        } else if (choice == 3) {
            int num;
            cout << "Enter a number: ";
            cin >> num;
            cout << num << " is a Palindrome number: " << is_palindrome_number(num) << endl;
        } else if (choice == 4) {
            int num;
            cout << "Enter a number: ";
            cin >> num;
            cout << num << " is a Harshad number: " << is_harshad_number(num) << endl;
        } else if (choice == 5) {
            int num;
            cout << "Enter a number: ";
            cin >> num;
            cout << num << " is a Krishnamurthy number: " << is_krishnamurthy_number(num) << endl;
        } else if (choice == 6) {
            int num;
            cout << "Enter a number: ";
            cin >> num;
            cout << num << " is an Evil number: " << is_evil_number(num) << endl;
        } else if (choice == 7) {
            int num;
            cout << "Enter a number: ";
            cin >> num;
            cout << num << " is a Fascinating number: " << is_fascinating_number(num) << endl;
        } else if (choice == 8) {
            cout << "Exiting program." << endl;

        } else {
            cout << "Invalid choice. Please try again." << endl;
        }
    }
    return 0;
}
